st2client/st2client/commands/action.py

```diff
@@ -1062,20 +1063,31 @@ def run(self, args, **kwargs):
         exclude_attributes = ','.join(exclude_attributes)
         kwargs['exclude_attributes'] = exclude_attributes
 
-        return self.manager.query(limit=args.last, **kwargs)
+        result, count = self.manager.query(limit=args.last, **kwargs)
+        return (result, count)
 
     def run_and_print(self, args, **kwargs):
-        instances = format_wf_instances(self.run(args, **kwargs))
 
-        if not args.json and not args.yaml:
+        result, count = self.run(args, **kwargs)
+        instances = format_wf_instances(result)
+
+        if args.json or args.yaml:
+            self.print_output(reversed(instances), table.MultiColumnTable,
+                              attributes=args.attr, widths=args.width,
+                              json=args.json,
+                              yaml=args.yaml,
+                              attribute_transform_functions=self.attribute_transform_functions)
+
+        else:
             # Include elapsed time for running executions
             instances = format_execution_statuses(instances)
+            self.print_output(reversed(instances), table.MultiColumnTable,
+                              attributes=args.attr, widths=args.width,
+                              attribute_transform_functions=self.attribute_transform_functions)
 
-        self.print_output(reversed(instances), table.MultiColumnTable,
-                          attributes=args.attr, widths=args.width,
-                          json=args.json,
-                          yaml=args.yaml,
-                          attribute_transform_functions=self.attribute_transform_functions)
+            if args.last >= self.default_limit and int(count) > args.last:
```

Comment:
It looks like `query` method returns None` if response doesn't contain X-Total header so this `int(count)` would fail.

So probably need to do something along the lines of `and count is not None and int(count)`...